lib/sparse/src/index/inverted_index/inverted_index_ram.rs

```diff
@@ -15,6 +18,29 @@ impl InvertedIndexRam {
     pub fn get(&self, id: &DimId) -> Option<&PostingList> {
         self.postings.get((*id) as usize)
     }
+
+    /// Upsert a vector into the inverted index.
+    pub fn upsert(&mut self, id: PointOffsetType, vector: SparseVector) {
+        for (index, dim_id) in vector.indices.into_iter().enumerate() {
+            let dim_id = dim_id as usize;
+            let weight = vector.weights[index];
+            match self.postings.get_mut(dim_id) {
+                Some(posting) => {
+                    // update existing posting list
+                    let posting_element = PostingElement::new(id, weight);
+                    posting.upsert(posting_element);
+                }
+                None => {
+                    // initialize new posting for dimension
+                    let mut posting_builder = PostingBuilder::new();
```

Comment:
I see there's only one posting element here. Do we actually need a builder here? `max_next_weight` is obviously `f32::NEG_INFINITY` which is defined as `DEFAULT_MAX_NEXT_WEIGHT`